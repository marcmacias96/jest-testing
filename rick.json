{
  "id": 1,
  "name": "Rick Sanches",
  "status": "alive",
  "specie": "humano",
  "sexo": "hombre"
}
