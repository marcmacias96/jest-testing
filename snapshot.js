export const getCharacter = data => ({
  id: data.id,
  name: data.name,
  status: data.status,
  species: data.species,
  sexo: data.sexo
});
