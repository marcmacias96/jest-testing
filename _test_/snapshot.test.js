import { getCharacter } from "../snapshot";
import rick from "../rick.json";

describe("Prueba de snapshots ", () => {
  test("Snapshot", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });
  test("Siempre fallara la instantanea", () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20)
    };
    expect(user).toMatchSnapshot();
  });
  test("Tenemos una excepcion dentro del codigo", () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Oscar Bargas"
    };
    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    });
  });
});
