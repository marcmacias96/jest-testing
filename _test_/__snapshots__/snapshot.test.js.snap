// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de snapshots  Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2019-11-10T03:16:11.722Z,
  "id": 7,
}
`;

exports[`Prueba de snapshots  Snapshot 1`] = `
Object {
  "id": 1,
  "name": "Rick Sanches",
  "sexo": "hombre",
  "species": undefined,
  "status": "alive",
}
`;

exports[`Prueba de snapshots  Tenemos una excepcion dentro del codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Oscar Bargas",
}
`;
